2996_reflection/reflection.md

```diff
@@ -996,6 +997,86 @@ Thus, the only instantiations of `TU_Ticket::Helper` occur because of the call t
 
 [On Compiler Explorer](https://godbolt.org/z/1vEjW4sTr).
 
+## Emulating typeful reflection
+Although we believe a single opaque `std::meta::info` type to be the best and most scalable foundation for reflection, we acknowledge the desire expressed by SG7 for future support for "typeful reflection". The following demonstrates one possible means of assembling a typeful reflection library, in which different classes of reflections are represented by distinct types, on top of the facilities proposed here.
+
+::: bq
+```cpp
+// Represents a 'std::meta::info' constrained by a predicate.
+template <std::meta::info Pred>
+  requires (type_of(reflect_invoke(Pred, {reflect_value(^int)})) == ^bool)
```

Comment:
Why do you need `reflect_invoke` here if `Pred` is already a constant? Can't this just be `decltype([: Pred:](^int))`?